File: Java_Programs/Fibonacci.java
import java.util.Scanner;

public class Fibonacci {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter N (>=1): ");
        if (!sc.hasNextInt()) { System.err.println("Invalid input"); sc.close(); System.exit(1); }
        int n = sc.nextInt();
        if (n < 1) { System.err.println("N must be >= 1"); sc.close(); System.exit(1); }
        long a = 0, b = 1;
        System.out.print("Fibonacci:");
        for (int i = 0; i < n; i++) {
            System.out.print(" " + a);
            long t = a + b; a = b; b = t;
        }
        System.out.println();
        sc.close();
    }
}

File: C_Programs/fibonacci.c
#include <stdio.h>

int main(void) {
    int n;
    printf("Enter N (>=1): ");
    if (scanf("%d", &n) != 1 || n < 1) {
        fprintf(stderr, "Invalid input\n");
        return 1;
    }
    long long a = 0, b = 1;
    printf("Fibonacci:");
    for (int i = 0; i < n; ++i) {
        printf(" %lld", a);
        long long t = a + b;
        a = b;
        b = t;
    }
    printf("\n");
    return 0;
}
